### PR TITLE
[ui] Tests for Sentinel Policies

### PR DESCRIPTION
**Pull request by philrenaud**
Acceptance tests on Sentinel Policy CRUD and 3 new Visual Diff / Snapshot tests (Sentinel Policies index page, Gallery page, and New Policy page). Largely borrowed from Roles/Policies acceptance tests and visual diff tests

### CODE REVIEW

File: ui/app/components/sentinel-policy-editor.hbs [hunk 1]
```diff
@@ -56,21 +56,21 @@
...
-			data-test-token-type="soft-mandatory"
+			data-test-enforcement-level="soft-mandatory"
 		as |F|>
 			<F.Label>Soft Mandatory</F.Label>
 		</G.Radio::Field>
 		<G.Radio::Field
 			@id="hard-mandatory"
 			checked={{eq @policy.enforcementLevel "hard-mandatory"}}
-			data-test-token-type="hard-mandatory"
+			data-test-enforcement-level="hard-mandatory"
```
Comment (philrenaud) [inline ui/app/components/sentinel-policy-editor.hbs:73]: The original templates was copied from ACL policy/token templates; this fixes the test-layer-only attributes.

Note: attributes that begin with `data-test-` are automatically stripped out for non-test builds.

File: ui/app/templates/administration/sentinel-policies/gallery.hbs [hunk 2]
```diff
@@ -16,10 +16,12 @@
...
-      <G.RadioCard class="form-container" @layout="fixed" @maxWidth="30%" @checked={{eq template.name
-        this.selectedTemplate}} id={{template.name}} data-test-template-card={{template.name}} {{on "change"
-        this.onChange}} as |R|>
-        <R.Label data-test-template-label>{{template.displayName}}</R.Label>
+      <G.RadioCard class="form-container" @layout="fixed" @maxWidth="30%"
+        @checked={{eq template.name this.selectedTemplate}}
+        id={{template.name}}
+        {{on "change" this.onChange}}
+      as |R|>
+        <R.Label data-test-template-card={{template.name}} data-test-template-label>{{template.displayName}}</R.Label>
```
Comment (philrenaud) [inline ui/app/templates/administration/sentinel-policies/gallery.hbs:24]: Newlined for readability, and added a data-test attribute

File: ui/app/templates/components/global-header.hbs [hunk 3]
```diff
@@ -10,6 +10,7 @@
       data-test-header-gutter-toggle
       class="gutter-toggle"
       aria-label="menu"
+      role="img"
```
Comment (philrenaud) [inline ui/app/templates/components/global-header.hbs:13]: Newer versions of our test runner will yell at us if we don't have an appropriate role for items with `aria-label` attributes. Basically the rule is "Spans shouldn't have aria-labels, you can just put text in them!" but this long-existing couple of items have been used as image-replacements and the role is appropriate (this is the hamburger/menu toggle for small screens)

File: ui/mirage/config.js [hunk 4]
```diff
@@ -932,18 +932,32 @@ export default function () {
...
+    const { Name, Description, EnforcementLevel, Policy, Scope } = JSON.parse(
+      req.requestBody
+    );
     return server.create('sentinelPolicy', {
       name: Name,
       description: Description,
-      rules: Rules,
+      enforcementLevel: EnforcementLevel,
+      policy: Policy,
+      scope: Scope,
```
Comment (philrenaud) [inline ui/mirage/config.js:943]: Before, mirage was only equipped to handle a trite implementation of our POST sentinel policies endpoint. Now it's slightly less trite.

**Review by shoenig — APPROVED** [review]

**Pull request merged**